### DISCUSSION TITLE
编写规范建议

### DISCUSSION THREAD

**Discussion by AresConnor** [MCSLTeam/MCSL2#120] [功能建议] [answered]
模块与模块之间非不要不耦合（方便单元测试，问题定位，以及并行开发），尽量不要在外部直接调用_开头与__开头的方法（某些情况下会造成逻辑混乱）。

**Comment by AresConnor** [comment]
因为懒得写issue模版，所以就在这里提了，起因是因为forge installer功能异常，改动了forge installer的部分代码，以及在外使用了installer的protected方法，导致想要修改，添加forge installer功能时很困难，需要翻阅其他模块（比如这个configPage），调试也花很久时间。因为这个原因，我认为是时候履行一些规范了，因为之后的开发 维护 和调试，都会因为这些规范而受益

**Comment by AresConnor** [comment]
另外 我想将publicFunctions.py命名为utils.py，这样或许更贴切一些

**Reply by AresConnor** [reply]
另外我希望落雪可以只使用forge installer的install方法。这个方法不需要手动的判断mc版本号的不同而引起的安装方式的变化。此外，这样方便我更新与维护installer基类下的所有类（包括forge installer，以及未来的fabric installer）

**Reply by AresConnor** [reply]
@LxHTT

**Comment by lxhtt** [comment, marked as answer]
差不多，后续还会继续解耦
